### DISCUSSION TITLE
MessageBag in livewire different or not in sync with the one exposed to Blade component?

### DISCUSSION THREAD

**Discussion by zack6849** [livewire/livewire#3046] [7. Bugs (archived)]
### Description

I'm not sure if this is a bug on livewire's part, or if i'm just doing something unexpected, feel free to close if this is entirely my mistake

So, a quick rundown

- I have a library of blade component classes I wrote, to avoid messy PHP code to check for errors in the blade, I do something like the below to add the is-invalid class to things with errors
```
/** @var ViewErrorBag $errors */
$errors = Session::get('errors', app(ViewErrorBag::class));
$this->default_attributes = [
    'class' => 'form-control', 
    'type' => 'text', 
    'name' => $name, 
    'id' => $id
];

if($errors->has($name)){
    debug("Appending is-invalid to class, error is set for $name");
    $this->default_attributes['class'] .= " is-invalid"; 
}
```

This has worked just fine, however, inside of a livewire component, for some reason, the message bag doesn't show anything for the line, despite @error($name) evaluating to true

I suspect this is because livewire uses it's own MessageBag, and the blade component is grabbing a separate one somehow
Is there a way to accomplish what i want?

Thanks,

**Context**
- Livewire version: 2.3.16
- Laravel version: 8.26.1
- Alpine version: N/A
- Browser: N/A

**Comment by joshhanley** [comment]
@zack6849 I'm not exactly clear how that is meant to work, so not sure where the issue would be arising. But like you suggested, I suspect it is this line `$errors = Session::get('errors', app(ViewErrorBag::class));` that's causing the issue.

If you are using Laravel's blade components, there was a directive added to the attribute bag called `class`, which allows conditional adding of classes, but requires Laravel 8.27.0. I use that now as I find it a lot cleaner.

https://laravel.com/docs/8.x/blade#conditionally-merge-classes

Hope this helps!

**Comment by zack6849** [comment]
@joshhanley  thanks for the reply, that sounds handy, however, i'm still stuck at not knowing how to determine if there's an error or not from my component short of @error, and I don't presume I can use that inside of another call, and even if i could, it'd look REALLY weird.

**Comment by joshhanley** [comment]
@zack6849 this is the code for my text input component `text-input.blade.php`, hopefully that helps. You don't need to get `$errors` yourself as it's already made available to blade views.

You can see the class method working here `->class(['border-red-500' => $errors->has($attributes->wire('model')->value)])`

```blade
@props([])

<div class="flex rounded-md shadow-sm">
    <input type="text"
        {{ $attributes->class(['border-red-500' => $errors->has($attributes->wire('model')->value)])->merge(['class' => 'flex-1 min-w-0 block w-full border-gray-300 focus:z-10 focus:ring-cyan-500 focus:border-cyan-500 disabled:bg-gray-100']) }}>
</div>
```
